### DISCUSSION TITLE
Function node for direct p5js code

### DISCUSSION THREAD

**Discussion by ReinaldoAssis** [peilingjiang/b5#236] [Q&A] [answered]
Hey there, I stumbled upon this repo and I found it fascinating, awesome work! I'd like to know if there are any plans to add a feature that enables creating new nodes with direct code (instead of using the function feature). Let me explain: 

I've already coded a simple inverted pendulum simulation in p5js (you can check it out [here](https://editor.p5js.org/reiziz/full/5ZuV7qG3s)) and the physical modeling of the system requires quite a bit of code, if I'd like to implement a node that simulates the inverted pendulum in b5 I'd have to do it visually and it would be nearly impossible/terribly tedious. If there were a feature that let me code the physical simulation directly and have it output, for example, the x position of the cart, the theta angle etc and connect it to other nodes that would be incredible. Is there already a way of doing this? Thanks.

**Comment by peilingjiang** [comment, marked as answer]
Unfortunately, you cannot do it now. But I agree it'll greatly increase the utility of the system. I'll find some time to work on it in the coming months, but I won't be able to make it supported soon.
